ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/ChainDB/Unit.hs

```diff
@@ -79,11 +80,35 @@ followerInstructionOnEmptyChain = do
     Left _      -> failWith $ "ChainDbError"
 
 
+-- | Test that a follower starts following a winning fork, even it was
+-- previously on a losing one.
+followerSwitchesToNewChain
+  :: (Block m ~ TestBlock, SupportsUnitTest m, MonadError TestFailure m) => m ()
+followerSwitchesToNewChain =
+  let fork i = TestBody i True
+  in do
+    b1 <- addBlock $ firstBlock 0 $ fork 0
+    b2 <- addBlock $ mkNextBlock b1 1 $ fork 0
+    f <- newFollower
+    void $ followerForward f [blockPoint b2]
+    b3 <- addBlock $ mkNextBlock b1 2 $ fork 1
+    void $ addBlock $ mkNextBlock b1 3 $ fork 1
+    followerInstruction f >>= \case
```

Comment:
Do you know where to find good examples?

Reply:
Given the chain we construct is in code, would it not be more error prone to provide a way to interpret that code in such a way it constructs this representation of a chain in case of a failure? It's marginally related but quickcheck-dynamic does something similar to provide an easy to past example failure given some formula: https://github.com/input-output-hk/quickcheck-dynamic/blob/main/quickcheck-dynamic/src/Test/QuickCheck/DynamicLogic/Internal.hs#L268

Reply:
> Do you know where to find good examples?


[Here is one](https://github.com/input-output-hk/ouroboros-network/blob/807ba41f849dbead8ffe5a9d73f5b77cafb5a53a/ouroboros-consensus/src/Ouroboros/Consensus/HardFork/History/EraParams.hs#L98), maybe there are better ones out there. For the case at hand what about adding something like:


```text
              f
              |
              v
G -- b1----- b2
      \
       \---- b3
        \
         \-- ???
```

I'm not sure about the `???` block actually